airflow/models/param.py

```diff
@@ -154,30 +170,34 @@ def __getitem__(self, key: str) -> Any:
         :param key: The key to fetch
         :type key: str
         """
-        param = super().__getitem__(key)
+        param = self.__dict[key]
         return param.resolve(suppress_exception=self.suppress_exception)
 
+    def get_param(self, key: str) -> Param:
+        """Get the internal :class:`.Param` object for this key"""
+        return self.__dict[key]
+
+    def items(self):
+        return ItemsView(self.__dict)
+
+    def values(self):
+        return ValuesView(self.__dict)
+
+    def update(self, *args, **kwargs) -> None:
+
+        if len(args) == 1 and isinstance(args[0], ParamsDict):
```

Comment:
```suggestion
        if len(args) == 1 and not kwargs and isinstance(args[0], ParamsDict):
```

If this is called with an invalid argument combination like `params.update({"b": 1}, c=2)`, we want to let the `super()` call catch it and emit an error.